django_tasks/backends/database/management/commands/db_worker.py

```diff
@@ -36,7 +36,7 @@ def __init__(
     def shutdown(self, signum: int, frame: Optional[FrameType]) -> None:
         if not self.running:
             logger.warning(
-                "Received %s - shutting down immediately.", signal.strsignal(signum)
+                "Received %s - terminating current task.", signal.strsignal(signum)
             )
             sys.exit(1)
```

Comment:
@mgax question for you. This implementation means that "immediately" actually marks the running task as failed before terminating. Is that a sensible idea, or should it terminate _harder_?

Reply:
I think that, ideally, we should be relying more on standard Python exception handling, instead of installing our own signal handler, because handling error states in reentrant functions is hard.

I've had a go at writing a task loop that waits for the current task to finish on the first CTRL-C, but kills it on the second CTRL-C. With this structure, it should be safe to attempt to save the task exit status, because that action, too, could be reliably interrupted by (yet another) CTRL-C, if needed.

https://gist.github.com/mgax/642c816b774a3b2bae4dcfac2d7d0d8b

Reply:
After the `sys.exit` call, if we removed our custom signal handlers, Python would be able to raise a `KeyboardInterrupt` on the 3rd ctrl-c, allowing the process to terminate much harder. 

I think I'm ok with the flow as currently implemented (counts of ctrl-c):

1. Terminate after the current task
2. Terminate the current task, then exit

2 is determinate on the task itself not handling `SystemExit` or `BaseException`, but that's probably ok. #9 will need a better way to hard terminate tasks, so this can be solved fully with that.